src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,13 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user)
+                .filter(u -> u != null)
+                .filter(u -> u.getAge() != null)
+                .map(u -> u.getAge())
+                .filter(age -> age >= 30)
+                .filter(age -> age <= 45)
+                .isPresent();
```

Comment:
`stream`을 정말 잘 사용해주셨네요!
```suggestion
        return Optional.ofNullable(user)
                .filter(u -> u != null)
                .filter(u -> u.getAge() != null)
                .map(User::getAge)
                .filter(User::isBetweenRange)
                .isPresent();
```

```java
private static boolean isBetweenRange(int age) {
    return 30 <= age && age <= 45;
}
```

이런식으로도 고칠 수 있을 것 같아보입니다!
물론 작성해주신 것이 잘못되었다는건 아닙니다. 잘 작성해주셨어요!

Reply:
오.. 넵넵 좋은 제안 감사합니다!